src/plugins/core/data_validation.ts

```diff
@@ -134,10 +134,12 @@ export class DataValidationPlugin
   }
 
   getValidationRuleForCell({ sheetId, col, row }: CellPosition): DataValidationRule | undefined {
-    for (const rule of this.rules[sheetId]) {
-      for (const range of rule.ranges) {
-        if (isInside(col, row, range.zone)) {
-          return rule;
+    if (this.rules[sheetId]) {
```

Comment:
With this appraoach , we'll end up with many many and more and more indentation levels. It hurts readability
Prefer an early return in those cases.
```ts
if (!this.rules[sheetId]) {
  return;
}
```

Reply:
Okay, noted!
Applied the change :+1: